src/plugins/data/public/ui/query_editor/query_editor.tsx

```diff
@@ -379,146 +321,259 @@ export default class QueryEditorUI extends Component<Props, State> {
-                options={{
-                  lineNumbers: 'on',
-                  lineHeight: 24,
-                  fontSize: 14,
-                  fontFamily: 'Roboto Mono',
-                  minimap: {
-                    enabled: false,
-                  },
-                  scrollBeyondLastLine: false,
-                  wordWrap: 'on',
-                  wrappingIndent: 'indent',
-                  lineDecorationsWidth: 0,
-                  lineNumbersMinChars: 2,
-                  wordBasedSuggestions: false,
-                }}
-                // suggestionProvider={{
-                //   provideCompletionItems: this.provideCompletionItems,
-                // }}
-              />
-            )}
-
-            <div
-              className={
-                !this.state.isCollapsed && useQueryEditor
-                  ? footerClassName
-                  : 'osdQueryEditorFooter-isHidden'
-              }
-            >
-              <EuiFlexGroup gutterSize="s" responsive={false}>
-                <EuiFlexItem grow={false}>{languageSelector}</EuiFlexItem>
-
-                <EuiFlexItem grow={false}>
-                  {this.state.lineCount} {this.state.lineCount === 1 ? 'line' : 'lines'}
-                </EuiFlexItem>
-                <EuiFlexItem grow={false}>
-                  {this.props.dataSet && `@${this.props.dataSet.timeFieldName}`}
-                </EuiFlexItem>
-              </EuiFlexGroup>
-            </div>
-          </EuiFlexItem>
-
-          {!this.state.isCollapsed && (
-            <EuiFlexItem grow={false}>
-              <div className="osdQueryEditor__filterBarWrapper">{this.props.filterBar}</div>
-            </EuiFlexItem>
-          )}
-        </EuiFlexGroup>
+      <div
+        className={classNames(
+          className,
+          'osdQueryEditor',
+          this.state.isCollapsed ? 'collapsed' : 'expanded',
+          !languageEditor.TopBar.Expanded && 'emptyExpanded'
+        )}
+      >
+        <div
+          ref={this.bannerRef}
+          className={classNames('osdQueryEditor__banner', this.props.bannerClassName)}
+        />
+        <div className="osdQueryEditor__topBar">
+          <QueryEditorBtnCollapse
+            onClick={() => this.setState({ isCollapsed: !this.state.isCollapsed })}
+            isCollapsed={!this.state.isCollapsed}
+          />
+          <div ref={this.props.dataSetContainerRef} className="osdQueryEditor__datasetPicker" />
+          <div className="osdQueryEditor__input">
+            {this.state.isCollapsed
+              ? languageEditor.TopBar.Collapsed()
+              : languageEditor.TopBar.Expanded && languageEditor.TopBar.Expanded()}
+          </div>
+          {languageSelector}
+          {this.props.queryActions}
+        </div>
+        {!this.state.isCollapsed && (
+          <div className="osdQueryEditor__body">{languageEditor.Body()}</div>
+        )}
+
+        {/*  <EuiFlexGroup gutterSize="xs" direction="column">
```

Comment:
Nit: why not just cleanup unused code?

Reply:
This refactor focussed on styling changes, but could not validate all the features the query bar intiially had such as Query assist and auto complete since they were in a partially reverted state. Preserved the previous code to make reincorporating them correctly easier. SHould be cleaned up in a followup PR